src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -509,17 +509,13 @@ internal void FlattenArgs(
             bool sourceFileSeen = false;
             bool optionsEnded = false;
 
-            var args = ArrayBuilder<string>.GetInstance();
-            args.AddRange(rawArguments);
-            args.ReverseContents();
-            var argsIndex = args.Count - 1;
-            while (argsIndex >= 0)
+            var argsEnumerator = rawArguments.GetEnumerator();
+            while (argsEnumerator.MoveNext())
```

Comment:
>while (argsEnumerator.MoveNext()) [](http://example.com/codeflow?start=12&length=33)

I would like to suggest an alternative approach.
1. Extract the body of this loop into a local function, let's call it "void processArg(string arg)". It should start with trimming the `arg`, and do what the body of the loop does, assuming that a `continue` is replaced with a `return`.
2. Change this loop to plain `foreach` over ```rawArguments```
3. Inside ```parseResponseFile```, instead of doing  ```notProcessedSplitList.Add(newArg);```, call ```processArg(newArg)```.